examples/eventsub_websocket/src/websocket.rs

```diff
@@ -1,170 +1,376 @@
     eventsub::{
         self,
-        event::websocket::{EventsubWebsocketData, ReconnectPayload, SessionData, WelcomePayload},
-        Event,
+        channel::{ChannelBanV1, ChannelUnbanV1},
+        event::websocket::{EventsubWebsocketData, ReconnectPayload, WelcomePayload},
+        Event, EventSubscription, Message, SessionData, Transport,
     },
-    types::{self},
-    HelixClient,
+    helix::{eventsub::CreateEventSubSubscription, ClientRequestError, HelixRequestPostError},
+    twitch_oauth2::{TwitchToken, UserToken},
+    types, HelixClient,
 };
-use twitch_oauth2::{TwitchToken, UserToken};
-
-pub struct WebsocketClient {
-    /// The session id of the websocket connection
-    pub session_id: Option<String>,
-    /// The token used to authenticate with the Twitch API
-    pub token: UserToken,
-    /// The client used to make requests to the Twitch API
-    pub client: HelixClient<'static, reqwest::Client>,
-    /// The user id of the channel we want to listen to
-    pub user_id: types::UserId,
-    /// The url to use for websocket
-    pub connect_url: url::Url,
-    pub opts: Arc<crate::Opts>,
+
+/// Connect to the websocket and return the stream
+///
+/// eventsub websocket doesn't support outgoing messages except pongs (which are implicitly handled by tungstenite)
+/// so we return only the receiving end of the socket
+///
+/// [Getting Events Using WebSockets](https://dev.twitch.tv/docs/eventsub/handling-websocket-events/)
+async fn connect(
+    request: impl IntoClientRequest + Unpin,
+) -> Result<SplitStream<WebSocketStream<MaybeTlsStream<tokio::net::TcpStream>>>, eyre::Error> {
+    tracing::info!("connecting to twitch");
+    let config = Some(WebSocketConfig {
+        max_message_size: Some(64 << 20), // 64 MiB
+        max_frame_size: Some(16 << 20),   // 16 MiB
+        accept_unmasked_frames: false,
+        ..WebSocketConfig::default()
+    });
+    let socket = tokio_tungstenite::connect_async_with_config(request, config, false)
+        .await
+        .context("Can't connect")?
+        .0
+        .split()
+        .1;
+
+    Ok(socket)
 }
 
-impl WebsocketClient {
-    /// Connect to the websocket and return the stream
-    pub async fn connect(
-        &self,
-    ) -> Result<
-        tokio_tungstenite::WebSocketStream<
-            tokio_tungstenite::MaybeTlsStream<tokio::net::TcpStream>,
-        >,
-        eyre::Error,
-    > {
-        tracing::info!("connecting to twitch");
-        let config = tungstenite::protocol::WebSocketConfig {
-            max_message_size: Some(64 << 20), // 64 MiB
-            max_frame_size: Some(16 << 20),   // 16 MiB
-            accept_unmasked_frames: false,
-            ..tungstenite::protocol::WebSocketConfig::default()
-        };
-        let (socket, _) =
-            tokio_tungstenite::connect_async_with_config(&self.connect_url, Some(config), false)
-                .await
-                .context("Can't connect")?;
+async fn refresh_if_expired(
```

Comment:
This always returns `Ok` - remove this and inline the TODO comment where needed.